### PR TITLE
Update `MediaQuery` documentation to highlight asynchronous nature of size

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This updates the documentation for `MediaQuery` and `Overlay` to include the following:

- That `MediaQueryData.size` is set asynchronously, and doesn't necessarily reflect the size of the current frame.
- That the specific `MediaQuery.sizeOf` and related methods are preferred over the more generic `MediaQuery.of`.
- That the size of the `Overlay` isn't necessarily the size returned by `MediaQueryData.size`

As well as adding some symbol links, and fixing some error in documentation.

## Tests
 - Documentation only

### CODE REVIEW

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 1]
```diff
@@ -1210,42 +1229,53 @@ class MediaQuery extends InheritedModel<_MediaQueryAspect> {
...
   /// throws an exception, if no such ancestor exists.
   ///
+  /// {@template flutter.widgets.media_query.sizeOf.asyncNote}
+  /// This returns the size of the media in logical pixels (e.g, the size of the
+  /// screen) of a previously rendered frame.
+  ///
+  /// Because the information is delivered asynchronously, it is not
+  /// deterministic which previous frame supplied the size returned by [size].
+  /// The initial size is the size of the created [FlutterView].
+  /// {@endtemplate}
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/media_query.dart:1242]: I think we should remove the async note for now. This will become relevant in a multi window world again, but we haven't fully decided how to implement this yet. It is possible that we will fake the size to be correct.

For now, you can actually assume that `size` is the size the frame will be rendered at.

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 2]
```diff
@@ -254,34 +254,47 @@ class MediaQueryData {
     return scaleFactor == 1.0 ? TextScaler.noScaling : TextScaler.linear(scaleFactor);
   }
 
-  /// The size of the media in logical pixels (e.g, the size of the screen).
+  /// The size of the media in logical pixels (e.g, the size of the screen) of a
+  /// previously rendered frame.
+  ///
+  /// Because the information is delivered asynchronously, it is not
+  /// deterministic which previous frame supplied the size returned by [size].
+  /// The initial size is the size of the created [FlutterView].
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/media_query.dart:262]: Let's undo this change, see note below.

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 3]
```diff
@@ -854,26 +867,26 @@ class MediaQueryData {
 
 /// Establishes a subtree in which media queries resolve to the given data.
 ///
-/// For example, to learn the size of the current media (e.g., the window
-/// containing your app), you can use [MediaQuery.sizeOf]:
+/// For example, to learn the size of a recent frame on the current view (e.g.,
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/media_query.dart:870]: remove the "a recent frame on", see below.

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 4]
```diff
@@ -854,26 +867,26 @@ class MediaQueryData {
...
+/// the [FlutterView] containing your app), you can use [MediaQuery.sizeOf] :
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/media_query.dart:871]: nit:
```suggestion
/// the [FlutterView] containing your app), you can use [MediaQuery.sizeOf]:
```

**Review by gspencergoog — COMMENTED** [review]
I also looked for other instances of `MediaQuery.of` and fixed them up to point to the specific attribute values instead. PTAL

**Review by goderbauer — APPROVED** [review]
LGTM
